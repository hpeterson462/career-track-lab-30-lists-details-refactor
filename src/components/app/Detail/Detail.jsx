import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ character }) => {
  return (
    <section data-testid="character">
      <p>{character.name}</p>
      <p>{character.race}</p>
      <p>{character.gender}</p>
      <p>{character.wikiUrl}</p>
    </section>
  );
};

// Character.propTypes = {
//   character: PropTypes.array().isRequired
// }
// // name: PropTypes.string.isRequired,
// // race: PropTypes.string.isRequired,
// // gender: PropTypes.string.isRequired,
// // wikiUrl: PropTypes.string.isRequired,
// // ).isRequired
// // };

export default Character;
