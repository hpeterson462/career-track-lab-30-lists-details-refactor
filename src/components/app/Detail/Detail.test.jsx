import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Detail from './Detail';

describe('Detail component', () => {
  afterEach(() => cleanup());
  it('renders Detail', () => {
    const { asFragment } = render(<Detail
      name="Spot"
      race="dog"
      gender="female"
      wikiUrl="www.wikidog.com"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
