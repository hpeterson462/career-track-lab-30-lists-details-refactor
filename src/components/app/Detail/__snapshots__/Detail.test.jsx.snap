// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders Detail 1`] = `
<DocumentFragment>
  <p>
    Spot
  </p>
  <p>
    dog
  </p>
  <p>
    female
  </p>
  <p>
    www.wikidog.com
  </p>
</DocumentFragment>
`;
